diff -up pcp-3.6.3/configure.in.s390x pcp-3.6.3/configure.in
--- pcp-3.6.3/configure.in.s390x	2012-05-18 12:31:14.000000000 +0200
+++ pcp-3.6.3/configure.in	2012-05-18 12:31:47.000000000 +0200
@@ -1636,7 +1636,7 @@ if test $target_os = mingw
 then
     pcp_lib_dir=/local/bin
     pcp_lib32_dir=/local/bin
-elif test $target_os = linux -a \( $target_cpu = x86_64 -o $target_cpu = powerpc64 \) -a $target_distro != debian
+elif test $target_os = linux -a \( $target_cpu = x86_64 -o $target_cpu = powerpc64 -o $target_cpu = s390x \) -a $target_distro != debian
 then
     # Linux on x86_64 and ppc64 puts libraries in .../lib64 (and rpmbuild
     # enforces that convention) but $libdir lies about it.
