diff --git a/src/pmcd/pmcd.defaults b/src/pmcd/pmcd.defaults
index 83ee4d9a9..56e125553 100644
--- a/src/pmcd/pmcd.defaults
+++ b/src/pmcd/pmcd.defaults
@@ -4,7 +4,7 @@
 # Behaviour regarding listening on external-facing interfaces;
 # unset PMCD_LOCAL to allow connections from remote hosts.
 # A value of 0 permits remote connections, 1 permits local only.
-# PMCD_LOCAL=1
+PMCD_LOCAL=1
 
 # Max length to which the queue of pending connections may grow
 # A value of 5 is the default.
diff --git a/src/pmlogger/pmlogger.defaults b/src/pmlogger/pmlogger.defaults
index 5462403ec..261a79039 100644
--- a/src/pmlogger/pmlogger.defaults
+++ b/src/pmlogger/pmlogger.defaults
@@ -4,7 +4,7 @@
 # Behaviour regarding listening on external-facing interfaces;
 # unset PMLOGGER_LOCAL to allow connections from remote hosts.
 # A value of 0 permits remote connections, 1 permits local only.
-# PMLOGGER_LOCAL=1
+PMLOGGER_LOCAL=1
 
 # Max length to which the queue of pending connections may grow
 # A value of 5 is the default.
